src/main/java/org/rumbledb/expressions/flowr/FlworVarDecl.java

```diff
@@ -36,8 +36,8 @@
     protected VariableReferenceExpression variableReferenceExpression;
     protected Expression expression;
 
-    // asSequence is null by default if the type of the variable in the for/let/groupBy clause is not specified.
-    protected FlworVarSequenceType asSequenceType;
+    // asSequence is never null.
```

Comment:
The field name seems out of date here. But in any case, I think we can remove this comment fully as the code throws an error if sequenceType is null in the constructor.

Reply:
done